kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala

```diff
@@ -0,0 +1,175 @@
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.KyuubiConf.ENGINE_TRINO_CONNECTION_CATALOG
+import org.apache.kyuubi.config.KyuubiConf.ENGINE_TRINO_CONNECTION_URL
+import org.apache.kyuubi.config.KyuubiConf.ENGINE_TRINO_MAIN_RESOURCE
+import org.apache.kyuubi.engine.ProcBuilder
+import org.apache.kyuubi.engine.trino.TrinoProcessBuilder._
+import org.apache.kyuubi.operation.log.OperationLog
+
+class TrinoProcessBuilder(
+    override val proxyUser: String,
+    override val conf: KyuubiConf,
+    val extraEngineLog: Option[OperationLog] = None) extends ProcBuilder with Logging {
+
+  override protected val executable: String = {
+    val trinoHomeOpt = env.get("TRINO_ENGINE_HOME").orElse {
+      val cwd = getClass.getProtectionDomain.getCodeSource.getLocation.getPath
+        .split("kyuubi-server")
+      assert(cwd.length > 1)
+      Option(
+        Paths.get(cwd.head)
+          .resolve("externals")
+          .resolve(module)
+          .toFile)
+        .map(_.getAbsolutePath)
+    }
+
+    trinoHomeOpt.map { dir =>
+      Paths.get(dir, "bin", TRINO_ENGINE_BINARY_FILE).toAbsolutePath.toFile.getCanonicalPath
+    }.getOrElse {
+      throw KyuubiSQLException("TRINO_ENGINE_HOME is not set! " +
+        "For more detail information on installing and configuring Trino, please visit " +
+        "https://kyuubi.apache.org/docs/stable/deployment/settings.html#environments")
+    }
+  }
+
+  override protected def mainResource: Option[String] = {
+    val jarName = s"${module}_$SCALA_COMPILE_VERSION-$KYUUBI_VERSION.jar"
+    // 1. get the main resource jar for user specified config first
+    conf.get(ENGINE_TRINO_MAIN_RESOURCE).filter { userSpecified =>
+      // skip check exist if not local file.
+      val uri = new URI(userSpecified)
+      val schema = if (uri.getScheme != null) uri.getScheme else "file"
+      schema match {
+        case "file" => Files.exists(Paths.get(userSpecified))
+        case _ => true
+      }
+    }.orElse {
+      // 2. get the main resource jar from system build default
+      env.get(KyuubiConf.KYUUBI_HOME)
+        .map { Paths.get(_, "externals", "engines", "trino", "jars", jarName) }
+        .filter(Files.exists(_)).map(_.toAbsolutePath.toFile.getCanonicalPath)
+    }.orElse {
+      // 3. get the main resource from dev environment
+      Option(Paths.get("externals", module, "target", jarName))
+        .filter(Files.exists(_)).orElse {
+          Some(Paths.get("..", "externals", module, "target", jarName))
+        }.map(_.toAbsolutePath.toFile.getCanonicalPath)
+    }
+  }
+
+  override protected def module: String = "kyuubi-trino-engine"
+
+  override protected def mainClass: String = "org.apache.kyuubi.engine.trino.TrinoSqlEngine"
+
+  override protected def childProcEnv: Map[String, String] = conf.getEnvs +
+    ("TRINO_ENGINE_JAR" -> mainResource.get)
+
+  override protected def commands: Array[String] = {
+    val buffer = new ArrayBuffer[String]()
```

Comment:
How about we pass all the configurations through normal java options?

we can simply get all the stuff inside the trino engine by single `new KyuubiConf()` only. besides, we can same a scopt dependency at engine side

Reply:
Done.